app/model/entity/Group.php

```diff
@@ -306,6 +306,24 @@ public function isAdminOrSupervisorOfSubgroup(User $user): bool {
     return false;
   }
 
+  /**
+   * User is admin of a group or any of its parents
+   * @param User $user
+   * @return bool
+   */
+  public function isAdminOfSuperGroup(User $user): bool {
+    $group = $this;
```

Comment:
There is no need to go through all parent groups again... there is `getAdminsIds()` method

Reply:
That would work too. Do you think it's worth rewriting it?

Reply:
Well... I think so, if I am not mistaken `isAdminOf()` method is using `getAdminsIds()` method so... you are doing searching which is not really necessary...

Reply:
You mean `getAdminIds` doesn't do the searching under the hood?

Reply:
Look in the code please...